paddle/gserver/gradientmachines/RecurrentGradientMachine.cpp

```diff
@@ -1399,8 +1399,8 @@ void RecurrentGradientMachine::createDataOutlinkCopySizeInfo(
              getBeamSize() > 1 ? finalPaths_.size() : finalPaths_[0].size());
     int* starts = inputSeqStartPos->getMutableData(false);
     int seqId = 0;
-    for (int i = 0; i < finalPaths_.size(); ++i) {
-      for (int j = 0; j < finalPaths_[i].size(); ++j) {
+    for (size_t i = 0; i < finalPaths_.size(); ++i) {
+      for (size_t j = 0; j < finalPaths_[i].size(); ++j) {
```

Comment:
do not quit understand why it's not good to use size_t in a for loop. Instead it writes:
` When appropriate, you are welcome to use standard types like size_t and ptrdiff_t.`

and here has some other explanation.
https://stackoverflow.com/questions/1951519/when-should-i-use-stdsize-t